templates/@theme-base/_full.scss

```diff
@@ -103,14 +110,20 @@ details:first-child {
   border-color: lighten($base, 10%);
 }
 
+// Email body background, text and links
 .email.message-body-container {
   [bgcolor], [style*="background"] {
     background-color: transparent !important;
-    div, p, 
-    h1, h2, h3,
-    h4, h5, h6 {
+
+    * {
       color: $text_color !important;
     }
+    a, a * {
+      color: $highlight !important;
+    }
+    a:active, a:focus, a:hover, a:active *, a:focus *, a:hover * {
+      color: lighten($highlight, 5%) !important;
+    } 
   }
 }
```

Comment:
Yes, it solves the problem with some formatted emails like newsletters, where elements can get nested like `table > tr > td > a > span`. The `span` would in this case overwrite the styling of each `a` element.

It would take a lot of whitelists and I'm assuming we want to style each element that uses the `color` property, right? :)

Reply:
I see, thanks for clarifying! The message panel is one of the aspects that I've found most challenging to get right so anything that helps here is great.

Reply:
It's hard to know what's right to do. Because the color of the text could be seen as part of the content.

E.g. if someone marks important parts of an email with the color red, then we don't want to overwrite this with the default text color, right? 

> 
> Ideally there should be a function that converts the text color to something that fits nice with the background for better readability/contrast. In theory this is possible using a scss function that makes an attribute selector for every possible HEX or rgb color (taking spacing into account), something like this:
> 
> ```
> [style*="color:#FF0000"], [style*="color: #FF0000"], [style*="color: rgb(255,0,0)"], [style*="color:rgb(255,0,0)"] {
>   color: $slightlydifferentred; // example
> }
> ```
> 
> This would result in 16.777.216 different groups of‬ selectors+declarations... So nevermind, unless we can use Javascript this is no option.


Another approach is to add a blend mode, like `mix-blend-mode: difference;`. That way the colors are always high contrast (and thus readable) while the user is able to make a distinction between different pieces of text. What do you think?

Reply:
I hear you. I have found the same issue when trying to keep the email contents pane with theme colours.

I guess one critical question is if theme colors should be applied to that pane when a newsletter or custom email has its own styling in the first place. Applying the theme palette to newsletters with custom styling might be too opinionated and lead to a larger task of matching trying to override all `[style]` attributes without having unwanted behavior.

On the other hand, it would be good to apply theme colors when emails are text based, but I need to investigate as I'm unsure of how and if it is possible to differentiate between them.